Fix the incompatible replay dialogue translations

abcdefg30 opened the pull request:
Updates Linguini to include the changes from https://github.com/Ygg01/Linguini/pull/24, and adds some new lines to stop the text from overflowing over the dialogue borders:
![grafik](https://user-images.githubusercontent.com/7704140/191269636-09f01d7e-269b-4324-aca0-8e04d27def4e.png)
![grafik](https://user-images.githubusercontent.com/7704140/191269645-0507712b-fc18-49b9-95f6-27536890d6a0.png)
![grafik](https://user-images.githubusercontent.com/7704140/191269656-792c76ed-85ec-496f-a47b-e3067be7378c.png)
![grafik](https://user-images.githubusercontent.com/7704140/191269981-958d808d-661c-4744-8781-3154250ad5ab.png)

abcdefg30 committed:
Update Linguini to 0.3.1

abcdefg30 committed:
Add line breaks to the incompatible replay translations

PunkPun commented on mods/common/languages/en.ftl:
```diff
@@ -502,11 +502,16 @@ replay-deletion-failed = Failed to delete replay file '{ $file }'. See the debug
+incompatible-replay-unknown-mod = { -incompatible-replay-recorded } an unknown mod:
+    { $mod }.
+incompatible-replay-unavailable-mod = { -incompatible-replay-recorded } an unavailable mod:
+    { $mod }.
```
Perhaps keep `$mod` in the same line? We use short names and it looks out of place to have them in a newline

RoosterDragon reviewed (dismissed):
Thanks for these changes, nice tidyup.

dragunoff commented on mods/common/languages/en.ftl:
```diff
@@ -502,11 +502,16 @@ replay-deletion-failed = Failed to delete replay file '{ $file }'. See the debug
-incompatible-replay-unavailable-mod = { -incompatible-replay-recorded } an unavailable mod: { $mod }.
-incompatible-replay-incompatible-version = { -incompatible-replay-recorded } an incompatible version: { $version }.
-incompatible-replay-unavailable-map = { -incompatible-replay-recorded } an unavailable map: { $map }.
+incompatible-replay-unknown-version = { -incompatible-replay-recorded } an unknown version:
+    { $version }.
+incompatible-replay-unknown-mod = { -incompatible-replay-recorded } an unknown mod:
+    { $mod }.
```
In these cases `$version` and `$mod` are `null` so the game crashes:
```
Exception of type `System.NullReferenceException`: Object reference not set to an instance of an object.
   at OpenRA.TranslationExts.ToFluentType(Object value) in /home/ivaylo/code/OpenRA/OpenRA.Game/TranslationExts.cs:line 41
   at OpenRA.Translation.TryGetString(String key, String& value, IDictionary`2 arguments) in /home/ivaylo/code/OpenRA/OpenRA.Game/Translation.cs:line 100
   at OpenRA.Mods.Common.Widgets.ConfirmationDialogs.ButtonPrompt(ModData modData, String title, String text, Dictionary`2 titleArguments, Dictionary`2 textArguments, Action onConfirm, Action onCancel, Action onOther, String confirmText, String cancelText, String otherText) in /home/ivaylo/code/OpenRA/OpenRA.Mods.Common/Widgets/ConfirmationDialogs.cs:line 43
   at OpenRA.Mods.Common.Widgets.Logic.ReplayUtils.IncompatibleReplayDialog(String text, Dictionary`2 textArguments, ModData modData, Action onCancel) in /home/ivaylo/code/OpenRA/OpenRA.Mods.Common/Widgets/Logic/ReplayUtils.cs:line 78
   at OpenRA.Mods.Common.Widgets.Logic.ReplayUtils.PromptConfirmReplayCompatibility(ReplayMetadata replayMeta, ModData modData, Action onCancel) in /home/ivaylo/code/OpenRA/OpenRA.Mods.Common/Widgets/Logic/ReplayUtils.cs:line 62
   at OpenRA.Mods.Common.Widgets.Logic.ReplayBrowserLogic.WatchReplay() in /home/ivaylo/code/OpenRA/OpenRA.Mods.Common/Widgets/Logic/ReplayBrowserLogic.cs:line 771
```

We don't need to pass any arguments - the mod/version is simply unknown.

dragunoff commented on mods/common/languages/en.ftl:
```diff
@@ -502,11 +502,16 @@ replay-deletion-failed = Failed to delete replay file '{ $file }'. See the debug
+    { $mod }.
+incompatible-replay-incompatible-version = { -incompatible-replay-recorded } an incompatible version:
+    { $version }.
+incompatible-replay-unavailable-map = { -incompatible-replay-recorded } an unavailable map:
+    { $map }.
```
I wasn't able to trigger this one, the game tries to download the map from the resource center and if it fails it won't even allow the "Watch" button to be clicked.

dragunoff replied:
Ah, I guess this is for when we load directly into a replay.

dragunoff reviewed (changes requested):
I think it makes more sense for the button to say "OK" instead of "Cancel":
![image](https://user-images.githubusercontent.com/1355810/191976586-83723167-6351-4421-b31a-4e7eecd02053.png)

Sorry for the scope creeping but I noticed some issues around this.

abcdefg30 committed:
Fix a crash when version or mod of a replay are null

abcdefg30 committed:
Change the incompatible replay dialogue text from "Cancel" to "OK"

abcdefg30 commented:
Updated:
- Changed "Cancel" to "OK"
![grafik](https://user-images.githubusercontent.com/7704140/192519897-57573c64-e8df-4e47-bb3d-813afafb0849.png)
- Removed line breaks for mod names
![grafik](https://user-images.githubusercontent.com/7704140/192519907-fe726f96-a3b9-4b85-931f-5fef1f521415.png)
- Fixed the crash with null arguments
![grafik](https://user-images.githubusercontent.com/7704140/192520084-1ad9ee08-1372-490b-9d88-0d04ddfe28f0.png)

PunkPun commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/6cd6fa16ef828d492415130e5bafe0ae61d1957a...607e3b63393dec981f1ba65c6675029adb37ab0d)
